tests/test_enterprise/api/test_views.py

```diff
@@ -9738,3 +9750,760 @@ def test_list_users_filtered(self):
+                        'content_key': enrollment_intention.content_key,
+                        'enterprise_customer': str(self.enterprise_customer.uuid),
+                        'course_key': enrollment_intention.course_key,
+                        'course_run_key': enrollment_intention.course_run_key,
+                        'is_course_run_enrollable': True,
+                        'applicable_enterprise_catalog_uuids': [fake_catalog_api.FAKE_CATALOG_RESULT.get('uuid')],
+                        'course_run_normalized_metadata': {
+                            'start_date': fake_catalog_api.FAKE_COURSE_RUN.get('start'),
+                            'end_date': fake_catalog_api.FAKE_COURSE_RUN.get('end'),
+                            'enroll_by_date': fake_catalog_api.FAKE_COURSE_RUN.get('seats')[1].get('upgrade_deadline'),
+                            'enroll_start_date': fake_catalog_api.FAKE_COURSE_RUN.get('enrollment_start'),
+                            'content_price': fake_catalog_api.FAKE_COURSE_RUN.get('first_enrollable_paid_seat_price'),
+                        },
+                        'created': enrollment_intention.created.strftime("%Y-%m-%dT%H:%M:%SZ"),
+                        'modified': enrollment_intention.modified.strftime("%Y-%m-%dT%H:%M:%SZ"),
+                        'has_existing_enrollment': False,
+                        'is_existing_enrollment_active': None,
+                        'is_existing_enrollment_audit': None,
+                    },
+                ],
+                'not_enrollable': [],
+            },
+            'already_enrolled': [],
+        }
+        assert response_data['metadata'] == {
+            'total_default_enterprise_course_enrollments': 1,
+            'total_needs_enrollment': {
+                'enrollable': 1,
+                'not_enrollable': 0,
+            },
+            'total_already_enrolled': 0,
+        }
+
+    @mock.patch('enterprise.content_metadata.api.EnterpriseCatalogApiClient')
+    @mock.patch('enterprise.models.utils.get_best_mode_from_course_key')
+    def test_default_enrollment_intentions_learner_status_nonstaff_lms_user_id_override(
+        self,
+        mock_get_best_mode_from_course_key,
+        mock_catalog_api_client
+    ):
+        """
+        Test default enterprise enrollment intentions for specific staff user linked to enterprise customer.
+        """
+        self.set_jwt_cookie(ENTERPRISE_LEARNER_ROLE, str(self.enterprise_customer.uuid))
+        enrollment_intention = self.create_mock_default_enterprise_enrollment_intention(mock_catalog_api_client)
+        mock_get_best_mode_from_course_key.return_value = VERIFIED_COURSE_MODE
+        factories.EnterpriseCustomerUserFactory(
+            user_id=self.user.id,
+            enterprise_customer=self.enterprise_customer,
+        )
+        query_params = (
+            f'enterprise_customer_uuid={str(self.enterprise_customer.uuid)}'
+            f'&lms_user_id=5'
```

Comment:
Is the idea that 5 will _not_ be the user id of `self.user`?  Maybe we can use `f'&lms_user_id={self.user.id + 1}` just to make it explicit that this test shows the requesting user can't get back data for a different user id if they are not staff?